### PR TITLE
Result Details Page

### PR DESCRIPTION
**Pull request by ChioreanOctavian**
https://jira.evozon.com/browse/PHPTRAININ-355

### CODE REVIEW

File: public/admin-results.phtml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <form method="POST">
+            <div class="form-group">
+                <label for="quizScore">Score</label>
+                <input
+                        name="score"
+                        type="number"
+                        class="form-control"
+                        id="quizScore"
+                        placeholder="Quiz score"
+                        min="0"
```
Comment (Raresmldvn) [inline public/admin-results.phtml:45]: Isn't there a `max` value as well?

File: src/Controllers/ResultController.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        if ($this->session->get('message')) {
+            $this->alertMessage($this->session->get('message'));
+            $this->session->delete('message');
+        }
+
+        return $this->renderer->renderView
+        (
+            'admin-results-listing.phtml',
+            $this->resultService->getResults($request)
+        );
```
Comment (Raresmldvn) [inline src/Controllers/ResultController.php:52]: available for all methods here: what if I'm not an admin and make this call from POSTMAN? (as a user/guest)

File: src/Services/ResultService.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     * @var PaginatorFactory
+     */
+    private $paginatorFactory;
+
+    /**
+     * @var RepositoryManagerInterface
+     */
+    private $repositoryManager;
+
+    public function __construct(
```
Comment (Raresmldvn) [inline src/Services/ResultService.php:29]: if you have docblocks everywhere else, add one for the constructor too

File: src/Services/ResultService.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        $questionRepo = $this->repositoryManager->getRepository(QuestionInstance::class);
+        //TODO: create function findByWithoutPagination
+        $questions =  $questionRepo->findBy
+        (
+            [
+                'quizInstanceId' => $quizId
+            ],
+            [],
+            0,
+            100
```
Comment (Raresmldvn) [inline src/Services/ResultService.php:121]: is 100 a default maximum of questions for a quiz? if it is, it should live in a constant

File: src/Services/ResultService.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        return $questionsDto;
+    }
+
+    /**
+     * @param int $questionId
+     * @return array
+     */
+    private function findAnswers(int $questionId): ?TextInstance
+    {
+        $answerRepo = $this->repositoryManager->getRepository(TextInstance::class);
```
Comment (Raresmldvn) [inline src/Services/ResultService.php:138]: abbrevations -> nope

File: src/Services/ResultService.php [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    /**
+     * @param int $quizId
+     * @return array
+     */
+    private function findQuestions(int $quizId): array
+    {
+        //TODO: create a service which return the questions of a quiz
+        $questionRepo = $this->repositoryManager->getRepository(QuestionInstance::class);
```
Comment (Raresmldvn) [inline src/Services/ResultService.php:112]: no abbrevations

File: src/Services/ResultService.php (hunk 4 above)
Comment (ChioreanOctavian) [reply src/Services/ResultService.php:121]: It will be implemented in next PR, working on it

File: src/Controllers/ResultController.php (hunk 2 above)
Comment (ChioreanOctavian) [reply src/Controllers/ResultController.php:58]: It will be implemented when autheticationService will be implemnted

**Pull request merged**